.github/workflows/analyze.yml

```diff
@@ -0,0 +1,68 @@
+# GH Action: Analyze
+# Runs various code analysis tools on the codebase
+name: analyze
+on: [push, pull_request]
+  
+jobs:
+  clang-format:
+    if: false # Disabled until a .clang-format style guide is established
+    name: clang-format
+    runs-on: ubuntu-latest
+    
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Download submodules
+      run: git submodule update --init --recursive
+
+    - name: Run style check with clang-format
+      uses: jidicula/clang-format-action@v4.5.0
+      with:
+        clang-format-version: '13'
+        check-path: 'rythe'
+
+  cppcheck:
+    name: cppcheck
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Download submodules
+      run: git submodule update --init --recursive
+
+    - name: Get cppcheck
+      run: DEBIAN_FRONTEND=noninteractive sudo apt-get update -yqq --allow-releaseinfo-change -o::Dpkg::Use-Pty=0 && DEBIAN_FRONTEND=noninteractive sudo apt-get install -yqq -o::Dpkg::Use-Pty=0 cppcheck
+
+    - name: cppcheck
+      run: cppcheck rythe/ --inline-suppr --suppress=syntaxError --xml --error-exitcode=1 2>&1 | python3 tools/cppcheck2github.py
```

Comment:
why suppress syntax errors?

Reply:
I don't actually know, these were the parameters set in the analyze action previously! :p

Reply:
welp ig we should ask @Algo-ryth-mix then XD